kalite/knowledgemap/templates/knowledgemap/knowledgemap.html

```diff
@@ -0,0 +1,616 @@
+{% extends base_template %}
+{% load i18n %}
+{% load staticfiles %}
+
+{% block practice_active %}active{% endblock practice_active %}
+
+{% block headcss %}{{ block.super }}
+    <link rel='stylesheet' type='text/css' href='{% static "css/distributed/tipsy.css" %}' />
+    <link rel='stylesheet' type='text/css' href='{% static "css/knowledgemap/knowledgemap.css" %}' />
+{% endblock headcss %}
+
+{% block headjs %}{{ block.super }}
+    <script src="{% static 'js/distributed/jquery.tipsy.js' %}"></script>
+
+    <script type="text/javascript">
+        var TOPIC_DATA_URLS = {
+            Domains: '{{ data_url }}/map_domains.json',
+            Subjects: '{{ data_url }}/map_subjects.json',
+            Topics: '{{ data_url }}/topics.json',
+            Tutorials: '{{ data_url }}/map_tutorials.json',
+            Exercises: '{{ data_url }}/exercises.json'
+        }
+        var ALL_TOPICS_URL = '{{ topics_url }}';
+    </script>
+{% endblock headjs %}
+
+{% block content %}
+<script src="http://d3js.org/d3.v3.min.js"></script>
+
+<script>
+window.statusModel.loaded.then(function() {
+
+  //add map explanation
+  var graph_explain_img=document.createElement("img");
+  graph_explain_img.style.position="absolute";
+  graph_explain_img.style.top="70px";
+  graph_explain_img.style.left="10px";
+  graph_explain_img.style.width="100px";
+  graph_explain_img.setAttribute('src', "{% static 'images/distributed/map_explain_img.png' %}");
+  document.body.appendChild(graph_explain_img);
+
+  var user_exercises_json;
+  var url = setGetParamDict("/api/knowledgemap/KnowledgeMapExerciselog/", {
+    user: statusModel.get("user_id"), 
+    completion_timestamp__gte: '2011-10-20',   //because the topic.json contain repetitive object under same parent, which cause problem for displaying knowledge map
+    completion_timestamp__lte: '2017-12-22'     //use this time filter to avoid encounter repetitive objects
```

Comment:
hehe, because after that time, there is no repeated topic in the json. But now we don't need to navigate via the map (now the map is static), so we can have repeated topic in the json